### ISSUE TITLE
new chost for arm breaks RUSTHOST detection

### ISSUE THREAD

**Issue by stefson** [smaeul/portage-overlay#12]
along with the new 17.0 profiles, there will be a change in the default CHOST for arm, to make life easier with clang and the likes. Now, when I read this I was pretty sure the new CHOST=armv7-unknown-linux-musleabihf would make a bit of trouble here and there with the rust ebuild you wrote so kindly. 

I'm sure it can be fixed, and I'd like to try if there is enough free time in the next weeks. However, I wonder how you'd like to see this getting fixed? The new CHOST is only active on a 17.0 profile, and fixing it for 17.0 would propably break the old 13.0 profile again, and an attempt to satisfy both of them propably my brain :)

**Comment by smaeul** [comment]
Would changing the first `RUSTARCH` case to:
```
	armv7*)
		RUSTARCH=armv7 ;;
```
And changing the second `RUSTLIBC` case to:
```
	arm*)
		RUSTLIBC=${CHOST##*-} ;;
```

work?

**Comment by stefson** [comment]
For armv7-unknown-linux-musleabihf, the config.toml should look like this. I edited it by hand and copied it into the ebuild on my local machine, just to test and make it work for now. So the good news is, that the chost change did not break the stages. I'm now going to test your suggestion. 

```
[llvm]
ninja = true
optimize = true
release-debuginfo = false
assertions = false
[build]
build = "armv7-unknown-linux-musleabihf"
host = ["armv7-unknown-linux-musleabihf"]
target = ["armv7-unknown-linux-musleabihf"]
cargo = "/var/tmp/portage/dev-lang/rust-1.23.0/work/stage0/bin/cargo"
rustc = "/var/tmp/portage/dev-lang/rust-1.23.0/work/stage0/bin/rustc"
docs = false
compiler-docs = false
submodules = false
python = "python2.7"
locked-deps = true
vendor = true
verbose = 2
[install]
prefix = "/usr"
libdir = "lib"
docdir = "share/doc/rust-1.23.0"
mandir = "share/rust-1.23.0/man"
[rust]
optimize = true
debuginfo = false
debug-assertions = false
use-jemalloc = false
default-linker = "armv7a-unknown-linux-musleabihf-gcc"
channel = "stable"
rpath = false
optimize-tests = true
[dist]
src-tarball = false
[target.armv7-unknown-linux-musleabihf]
cc = "armv7a-unknown-linux-musleabihf-gcc"
cxx = "armv7a-unknown-linux-musleabihf-g++"
linker = "armv7a-unknown-linux-musleabihf-gcc"
ar = "armv7a-unknown-linux-musleabihf-ar"

**Comment by stefson** [comment]
It works for both, CHOST=armv7-unknown-linux-musleabihf and CHOST=armv7-unknown-linux-gnueabihf.

**Comment by stefson** [comment]
Cargo needs the same fix, obviously. Speaking of fixes, the http-parser upgrade to :0/2.8.0 gave us a bit of trouble with the now outdated rust-1.23.0, because with :0/2.8.0 the inherited cargo binary needed for bootstrapping errors with missing symbol errors. Which is related to the abi change, and it was fixed for now with restraining the rust-1.23.0 ebuild to 0/2.6.2.

**Comment by smaeul** [comment]
Yeah, I need to figure out how to statically link cargo as part of a rust extended build. Then it won't be an issue at all

**Comment by stefson** [comment]
Take your time, it will not matter as long as the ebuilds are in tree. 

Will check tomorrow if the proposed fix breaks the builds on the 13.0 profile, which is still the default unfortunatley. If they do, I'd suggest to commit the fixes to all rust and cargo ebuilds, plus to fix http-parser dep to 0.2.6.2 for now. 

I'll open a pull request at the musl overlay? It needs a bit of attention anyway :-)

**Comment by stefson** [comment]
Your fix doesn't break the 13.0 profiles, it is good to go :)
